.github/workflows/prerelease.yml

```diff
@@ -43,100 +47,80 @@ jobs:
         env:
           GOPRIVATE: 'github.com/hashicorp/*'
         run: |
-          go install github.com/hashicorp/go-changelog/cmd/changelog-build@6ec9be372335f39c5df27e232c3669db7f5183a5
-          go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.49.0
+          go install github.com/hashicorp/go-changelog/cmd/changelog-build@522d403eacf1dca87cacec6dbc37fdfcda262d26
           go mod tidy
-          sudo wget https://github.com/jmespath/jp/releases/latest/download/jp-linux-amd64 -O /usr/local/bin/jp
-          sudo chmod +x /usr/local/bin/jp
-
-      - name: Run Unit Tests and Linter
-        run: make test-ci
-
-      - name: Upload Unit Test Coverage Artifact
-        uses: actions/upload-artifact@v3
-        with:
-          name: Test Coverage
-          path: coverage.html
+      
+      - name: Check For Changes
+        id: changes
+        run: |
+          CURRENT_VERSION=$(git describe --tags `git rev-list --tags --max-count=1`)
+          DIFF_CONTENT=$(git diff $CURRENT_VERSION origin/main)
+          if [[ $DIFF_CONTENT == "" ]]; then
+            echo "There were no changes since the last release."
+            echo "There were no changes since the last release." >> $GITHUB_STEP_SUMMARY
+            echo "SHOULD_RELEASE=false" >> $GITHUB_OUTPUT
+          else 
+            echo "There were changes since the last release."
+            echo "SHOULD_RELEASE=true" >> $GITHUB_OUTPUT
+            echo "CURRENT_VERSION=$CURRENT_VERSION" >> "$GITHUB_ENV"
+          fi
 
-      - name: Run E2E Tests
-        if: ${{ !inputs.skip-e2e }}
+      - name: Prepare Release
+        if: ${{ steps.changes.outputs.SHOULD_RELEASE == 'true' }}
```

Comment:
Why would one set `SHOULD_RELEASE` while not being on the main branch?

Reply:
SHOULD_RELEASE is probably more accurately named HAS_CHANGES. I'll update the naming